### PR TITLE
Update demo page and tests of the `BitModal` component for the new Position parameter (#1984)

### LINKED ISSUE #1984: Update demo page and tests of the `BitModal` component for the new `Position` parameter
After implementing the new `Position` parameter, it is required to update the demo page and tests of the `BitModal` component to reflect this parameter.

### PR DESCRIPTION
**Pull request by sajadkardel**
this closes #1984

### CODE REVIEW

File: src/Client/Web/Bit.Client.Web.BlazorUI/Components/Modal/BitModal.razor [hunk 1]
```diff
@@ -3,33 +3,26 @@
...
                     </div>
+                }
+                <div class="bit-mdl-container">
+                    <div class="bit-mdl-scrl-cnt">
+                        @ChildContent
+                    </div>
+                </div>
             </div>
         </div>
     </div>
```
Comment (msynk) [inline src/Client/Web/Bit.Client.Web.BlazorUI/Components/Modal/BitModal.razor:28]: please revert these changes, since it's not related to this PR.

File: src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Components/Modal/BitModalDemo.razor [hunk 2]
```diff
@@ -58,4 +58,60 @@
             </div>
         </ExamplePreview>
     </ComponentExampleBox>
+    <ComponentExampleBox Title="BitModal with position" HTMLSourceCode="@example2HTMLCode" CSharpSourceCode="@example1CSharpCode" ExampleId="example2">
+        <ExamplePreview>
+            <div class="example-desc">To set the modal position on the page you can use the BitModalPosition parameter.</div>
+            <div>
+                <BitButton OnClick=@(()=>IsOpenInPosition=true)>Open Modal in the top and right position</BitButton>
```
Comment (msynk) [inline src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Components/Modal/BitModalDemo.razor:65]: please add more options for the position settings

**Review by msynk — CHANGES_REQUESTED** [review]

**Comment by msynk** [comment]
@maryamhdr & @Cyrus-Sushiant 
plz review

File: src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Modal/BitModalTests.cs [hunk 3]
```diff
@@ -254,6 +254,67 @@ public void BitModalOnDismissShouldWorkCorrect(Visual visual)
...
+
+            DataRow(Visual.Material, BitModalPosition.Center),
+            DataRow(Visual.Material, BitModalPosition.TopLeft),
+            DataRow(Visual.Material, BitModalPosition.TopCenter),
+            DataRow(Visual.Material, BitModalPosition.TopRight),
+            DataRow(Visual.Material, BitModalPosition.CenterLeft),
+            DataRow(Visual.Material, BitModalPosition.CenterRight),
+            DataRow(Visual.Material, BitModalPosition.BottomLeft),
+            DataRow(Visual.Material, BitModalPosition.BottomCenter),
+            DataRow(Visual.Material, BitModalPosition.BottomRight)
```
Comment (maryamhdr) [inline src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Modal/BitModalTests.cs:286]: Why have you added visual as a parameter here?
Is there any problem with removing it?

Comment (sajadkardel) [reply src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Modal/BitModalTests.cs:286]: I think this is a parameter that we must follow in all tests (regardless of what property it is written for...) so that our tests work properly for all three different designs.
What do you think? @msynk @maryamhdr

**Review by Cyrus-Sushiant — DISMISSED** [review]

File: src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Modal/BitModalTests.cs (hunk 3 above)
Comment (ysmoradi) [reply src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Modal/BitModalTests.cs:286]: I'd recommend keep using it. Because it make sure components behavior remains the same no matter what visual is being used

Comment (msynk) [reply src/Client/Web/Bit.Client.Web.BlazorUI.Tests/Modal/BitModalTests.cs:286]: let's keep them for now 😉

File: src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Components/Modal/BitModalDemo.razor [hunk 4]
```diff
@@ -58,4 +58,111 @@
...
+                            varius. Quisque ac auctor ligula. Fusce fringilla, odio et dignissim iaculis, est lacus ultrices risus,
+                            vitae condimentum enim urna eu nunc. In risus est, mattis non suscipit at, mattis ut ante. Maecenas
+                            consectetur urna vel erat maximus, non molestie massa consequat. Duis a feugiat nibh. Sed a hendrerit diam,
+                            a mattis est. In augue dolor, faucibus vel metus at, convallis rhoncus dui.
+                        </p>
+                    </div>
+                </BitModal>
+            </div>
+        </ExamplePreview>
+    </ComponentExampleBox>
```
Comment (msynk) [inline src/Client/Web/Playground/Bit.Client.Web.BlazorUI.Playground/Web/Pages/Components/Modal/BitModalDemo.razor:167]: instead of adding a whole other component to introduce other available options, make the same component open with different values for the Position parameter.

**Review by msynk — CHANGES_REQUESTED** [review]

**Review by maryamhdr — APPROVED** [review]

**Review by Cyrus-Sushiant — APPROVED** [review]

**Review by msynk — APPROVED** [review]
